wolfssl/wolfcrypt/settings.h

```diff
@@ -694,8 +694,14 @@ extern void uITRON4_free(void *p) ;
     #endif
     /* FreeRTOS pvPortRealloc() implementation can be found here:
         https://github.com/wolfSSL/wolfssl-freertos/pull/3/files */
-    #if !defined(USE_FAST_MATH) || defined(HAVE_ED25519) || defined(WOLFSSL_ESPIDF)
-        #define XREALLOC(p, n, h, t) pvPortRealloc((p), (n))
+    #if !defined(USE_FAST_MATH) || defined(HAVE_ED25519)
+        #if defined(WOLFSSL_ESPIDF)
+            /*In IDF, realloc(p, n) is equivalent to 
+            heap_caps_realloc(p, s, MALLOC_CAP_8BIT) */
+            #define XREALLOC(p, n, h, t) realloc((p), (n))
```

Comment:
@miyazakh : If `WOLFSSL_ESPIDF` has `realloc` then why not use the standard `malloc`/`free` as well? Does it really need FreeRTOS `pvPortMalloc` and `vPortFree`?

Reply:
@dgarske : malloc() and free() are used in IDF. Those pvPortMalloc and vPortFree are mapped to malloc() and free() at [portable.h ](https://github.com/espressif/esp-idf/blob/132cc67c03364f5f032aa65c6b88f1f1bf7ecbee/components/freertos/include/freertos/portable.h#L149) of ported freertos in IDF.

Reply:
@miyazakh : I see, these are in the FREERTOS section of settings.h. This is fine then.